LLAP/llap.sh

```diff
@@ -0,0 +1,436 @@
+        -s '//configuration/property[last()]' -t elem -n "value" -v "/registry"\
+        -s '//configuration' -t elem -n "property" \
+        -s '//configuration/property[last()]' -t elem -n "desription" -v "" \
+        -s '//configuration/property[last()]' -t elem -n "name" -v "hadoop.registry.rm.enabled" \
+        -s '//configuration/property[last()]' -t elem -n "value" -v "true"\
+        /etc/hadoop/conf/core-site.xml
+    fi
+}
+
+##add missing log4j file on all nodes
+function get_log4j() {
+    echo "import missing log4j library..."
+    wget -nv --timeout=30 --tries=5 --retry-connrefused \
+    	https://repo1.maven.org/maven2/org/apache/logging/log4j/log4j-slf4j-impl/2.10.0/log4j-slf4j-impl-2.10.0.jar
+    cp log4j-slf4j-impl-2.10.0.jar /usr/lib/hive/lib
+}
+
+##repackage tez_lib_uris and place on HDFS; only need to do this on one node. 
+function package_tez_lib_uris(){
+	echo "repackage tez lib uris..."
+	cp /usr/lib/tez/lib/* /usr/lib/tez
+    cp /usr/local/share/google/dataproc/lib/* /usr/lib/tez
+	tar -czvf tez.tar.gz /usr/lib/tez
+	runuser -l hdfs -c 'hdfs dfs -mkdir /tez'
+    sleep 100
+	until `hdfs dfs -copyFromLocal tez.tar.gz /tez`; do echo "Retrying"; sleep 10; done
+}
+
+#reconfigure tez.lib.uris to point to hdfs rather than local filesytem; run on all nodes so we have compatiable config files
+function configure_tez_site_xml() {
+    echo "reconfigure tez-site.xml..."
+	sed -i 's@file:/usr/lib/tez,file:/usr/lib/tez/lib,file:/usr/local/share/google/dataproc/lib@${fs.defaultFS}/tez/tez.tar.gz@g' /etc/tez/conf/tez-site.xml
+}
+
+###add yarn service directory for hive; run only on one node
+function add_yarn_service_dir(){
+    echo "adding yarn service directory on the hive user..."
+    runuser -l hdfs -c 'hdfs dfs -mkdir /user/hive/.yarn'
+	runuser -l hdfs -c 'hdfs dfs -mkdir /user/hive/.yarn/package'
+	runuser -l hdfs -c 'hdfs dfs -mkdir /user/hive/.yarn/package/LLAP'
+	runuser -l hdfs -c 'hdfs dfs -chown hive:hive  /user/hive/.yarn/package/LLAP'
+}
+
+# All nodes need to run this. These files 
+function replace_core_llap_files() {
+	echo "replacing llap files..."
+	wget https://github.com/jster1357/llap/archive/refs/heads/main.zip -O main.zip
+	unzip main.zip
+	cp llap-main/package.py /usr/lib/hive/scripts/llap/yarn/package.py 
+	cp llap-main/runLlapDaemon.sh /usr/lib/hive/scripts/llap/bin/runLlapDaemon.sh
+    cp llap-main/llap_restart.sh /usr/lib/hive/scripts/llap/bin/llap_restart.sh 
+
+	#open missing properties files
+	cp /usr/lib/hive/conf/llap-cli-log4j2.properties.template /usr/lib/hive/conf/llap-cli-log4j2.properties
+	cp /usr/lib/hive/conf/llap-daemon-log4j2.properties.template /usr/lib/hive/conf/llap-daemon-log4j2.properties
+}
+
+##if the metadata value exists, then we want to configure the local ssd as a caching location.
+function configure_SSD_caching_worker(){
+    if [[ -n "$HAS_SSD" ]]; then
+        echo "ssd"
+        mkdir /mnt/1/llap
+        chmod 777 /mnt/1/llap
```

Comment:
Can we give a more restricted access instead of 777?

Reply:
I attempted to do that but I got consistent failures when I did so.